### ISSUE TITLE
no-on-calls-in-components recommended solution

### ISSUE THREAD

**Issue by karellm** [ember-cli/eslint-plugin-ember#223]
As a [ember-keyboard](https://github.com/patience-tema-baron/ember-keyboard), when I started a new app, the following triggered the `no-on-calls-in-components`:

```js
import { keyUp, keyDown } from 'ember-keyboard';

. . . .

aFunction: Ember.on(keyUp('KeyS'), function() {
  console.log('`KeyS` was pressed');
}),
```

There is a [ticket](https://github.com/patience-tema-baron/ember-keyboard/issues/79) there but I was wondering what was your recommandation about this issue. Should the linter be more specific and only be triggered when there is an alternative syntax?

**Comment by buschtoens** [comment]
From the rules docs:

> Prevents using `.on()` *in favour of component's lifecycle hooks*.

The way I read it, the intention is not to outright ban `on()` in components, but only ban it for component lifecycle hooks (`didInsertElement`, ...), because the order of execution is not really deterministic for `on()`.

AFAIK the [`@ember/object/evented` mixin](https://www.emberjs.com/api/ember/release/classes/Evented) is here to stay and has valid use cases (ember-keyboard for instance).

I don't have any say in this, but I'm pretty sure that a PR making the rule more specific, so that it only bans usage of `on()` with lifecycle hook event names, would be gladly accepted.

**Comment by karellm** [comment]
@buschtoens Thanks for answering. I currently have limited bandwidth to contribute to a new project but will keep this open in the event someone feel like tackling it.
